core/src/main/java/com/netease/arctic/io/ArcticHadoopFileIO.java

```diff
@@ -71,7 +71,9 @@ public void deleteFile(String path) {
       FileSystem fs = getFs(toDelete);
 
       try {
-        fs.delete(toDelete, false);
+        if (!fs.delete(toDelete, false)) {
+          throw new IOException("fileSystem delete " + path + " is failed");
+        }
```

Comment:
This modification will cause unit tests to fail on Windows according to @baiyangtx. Perhaps we should fix this problem in another PR, and this PR only fix the `OrphanFilesCleanService`.

Additionally, I check the Iceberg `HadoopFileIO`, it also doesn't throw an exception
```
  public void deleteFile(String path) {
    Path toDelete = new Path(path);
    FileSystem fs = Util.getFs(toDelete, hadoopConf.get());
    try {
      fs.delete(toDelete, false /* not recursive */);
    } catch (IOException e) {
      throw new RuntimeIOException(e, "Failed to delete file: %s", path);
    }
  }
```

Reply:
Yes, here, if the hdfs file deletion fails, false will be returned

Reply:
I deleted this code and resubmitted it.